ros2component/test/test_cli_list.py

```diff
@@ -0,0 +1,150 @@
+from launch.actions import ExecuteProcess
+from launch.actions import OpaqueFunction
+
+from launch_ros.actions import Node
+
+import launch_testing
+import launch_testing.asserts
+import launch_testing.markers
+import launch_testing.tools
+import launch_testing_ros.tools
+
+import pytest
+
+from rmw_implementation import get_available_rmw_implementations
+
+LIST_THREE_NODES = """\
+/ComponentManager
+  1  /listener
+  2  /talker
+  3  /talker
+"""
+
+
+# TODO(BMarchi): Opensplice doesn't get along with running any cli command with ExecuteProcess,
+# it just hangs there making `wait_for_timeout` fail. All tests should run fine with opensplice.
+@pytest.mark.rostest
+@launch_testing.parametrize(
+    'rmw_implementation',
+    [v for v in get_available_rmw_implementations() if v != 'rmw_opensplice_cpp'])
+def generate_test_description(rmw_implementation, ready_fn):
+    additional_env = {'RMW_IMPLEMENTATION': rmw_implementation}
+    component_node = Node(
+        package='rclcpp_components', node_executable='component_container', output='screen')
+    listener_command_action = ExecuteProcess(
+        cmd=['ros2', 'component', 'load', '/ComponentManager',
+             'ros2component_fixtures', 'ros2component_fixtures::Listener'],
+        additional_env={
+            'RMW_IMPLEMENTATION': rmw_implementation,
+            'PYTHONUNBUFFERED': '1'
+        },
+        name='ros2component-cli',
+        output='screen',
+        on_exit=[
+            ExecuteProcess(
+                cmd=['ros2', 'component', 'load', '/ComponentManager',
+                     'ros2component_fixtures', 'ros2component_fixtures::Talker'],
+                additional_env={
+                    'RMW_IMPLEMENTATION': rmw_implementation,
+                    'PYTHONUNBUFFERED': '1'
+                },
+                name='ros2component-cli',
+                output='screen',
+                on_exit=[
+                    ExecuteProcess(
+                        cmd=['ros2', 'component', 'load', '/ComponentManager',
+                             'ros2component_fixtures', 'ros2component_fixtures::Talker'],
+                        additional_env={
+                            'RMW_IMPLEMENTATION': rmw_implementation,
+                            'PYTHONUNBUFFERED': '1'
+                        },
+                        name='ros2component-cli',
+                        output='screen'
+                    )
+                ]
+            )
+        ]
+    )
+    return LaunchDescription([
+        # Always restart daemon to isolate tests.
+        ExecuteProcess(
+            cmd=['ros2', 'daemon', 'stop'],
+            name='daemon-stop',
+            on_exit=[
+                ExecuteProcess(
+                    cmd=['ros2', 'daemon', 'start'],
+                    name='daemon-start',
+                    on_exit=[
+                        component_node,
+                        OpaqueFunction(function=lambda context: ready_fn()),
```

Comment:
@BMarchi why isn't this the last action being called? Also, consider using `launch_testing.actions.ReadyToTest` instead (use of `ready_fn` is due for deprecation, eventually).

Reply:
For some reason the opaque function is working as expected meanwhile ReadyToTest throws me a time out for process launching.

Reply:
That's unexpected. `ReadyToTest` should be a drop-in replacement. Can you confirm it isn't? If so, open an issue in `launch` with a small repro.